File: java/PE047.java
public class PE047
{
    public static void main(String[] args)
    {
        for(int i = 1; ; i++) {
            int j;
            for(j = 0; j < 4; j++) {
                if(isFour(i+j))
                    continue;
                break;
            }
            if(j == 4) {
                System.out.println(i);
                break;
            }
        }
    }

    private static boolean isFour(int n)
    {
        int count = 0;
        for(int i = 2; n != 1; i++) {
            if(n%i == 0) {
                count++;
            }
            while(n%i == 0)
                n /= i;
        }
        if(count == 4)
            return true;
        return false;
    }
}

File: cpp/PE047.cpp
#include <iostream>
#include <cmath>
using namespace std;

bool isFour(int n)
{
    int i, j = 0;
    for(i = 2; i <= n; i++){
        if(n%i == 0){
            j++;
            n /= i;
        }
        while(n%i == 0)
            n /= i;
    }
    return (j == 4);
}
int main()
{
    int i, j;
    for(i = 2; ; i++){
        for(j = i; j < i+4; j++){
            if(!isFour(j))
                break;
        }
        if(j == i+4)
            break;
    }
    cout << i << endl;
    return 0;
}
